lib/nerve/reporter/zookeeper.rb

```diff
@@ -1,30 +1,64 @@
 require 'nerve/reporter/base'
+require 'thread'
 require 'zk'
 
+
 class Nerve::Reporter
   class Zookeeper < Base
+    @@zk_pool = {}
+    @@zk_pool_count = {}
+    @@zk_pool_lock = Mutex.new
+
     def initialize(service)
-      %w{zk_hosts zk_path instance_id host port}.each do |required|
+      %w{zk_hosts zk_path}.each do |required|
         raise ArgumentError, "missing required argument #{required} for new service watcher" unless service[required]
       end
-      @path = service['zk_hosts'].shuffle.join(',') + service['zk_path']
-      @data = parse_data({'host' => service['host'], 'port' => service['port'], 'name' => service['instance_id']})
+      # Since we pool we get one connection per zookeeper cluster
+      @path = service['zk_hosts'].sort.join(',')
```

Comment:
these variable names are not well-named now; for instance, the `@path` variable no longer contains anything path-related. maybe `@zk_hosts` ?